avivator/src/components/Controller/components/Slicer.js

```diff
@@ -1,42 +1,100 @@
 import React from 'react';
 import Grid from '@material-ui/core/Grid';
-
 import Slider from '@material-ui/core/Slider';
-
-import { useImageSettingsStore } from '../../../state';
+import Button from '@material-ui/core/Button';
+import { useImageSettingsStore, useViewerStore } from '../../../state';
+import { truncateDecimalNumber } from '../../../utils';
+import { EPSILON } from '../../../constants';
 
 const Slicer = () => {
-  const { setImageSetting, xSlice, ySlice, zSlice } = useImageSettingsStore();
+  const {
+    setClippingPlaneSettings,
+    sphericals: [spherical]
+  } = useImageSettingsStore();
   const sliceValuesAndSetSliceFunctions = [
-    [xSlice, v => setImageSetting({ xSlice: v }), 'x'],
-    [ySlice, v => setImageSetting({ ySlice: v }), 'y'],
-    [zSlice, v => setImageSetting({ zSlice: v }), 'z']
+    [
+      spherical.phi,
+      v => setClippingPlaneSettings(0, 'phi', v),
+      'ϕ',
+      [0, Math.PI * 2]
+    ],
+    [
+      spherical.theta,
+      v => setClippingPlaneSettings(0, 'theta', v),
+      'ϴ',
+      [-Math.PI / 2, Math.PI / 2]
+    ],
+    [
+      spherical.radius,
+      v => setClippingPlaneSettings(0, 'radius', v),
+      'r',
+      // Since the box has diagonal length (1 + 1)^{\frac{1}{2}}
+      [EPSILON, Math.sqrt(2)]
+    ]
   ];
-  return sliceValuesAndSetSliceFunctions.map(([slice, setSlice, label]) => (
-    <Grid
-      container
-      direction="row"
-      justify="flex-start"
-      alignItems="center"
-      key={label}
-    >
-      <Grid item xs={1} style={{ marginBottom: 8 }}>
-        {label}:
-      </Grid>
-      <Grid item xs={11}>
-        <Slider
-          value={slice}
-          onChange={(e, v) => setSlice(v)}
-          valueLabelDisplay="auto"
-          getAriaLabel={() => `${label} slider`}
-          min={0}
-          max={1}
-          step={0.005}
-          orientation="horizontal"
-        />
+  const slicers = sliceValuesAndSetSliceFunctions.map(
+    ([val, setVal, label, [min, max]]) => (
+      <Grid
+        container
+        direction="row"
+        justify="flex-start"
+        alignItems="center"
+        key={label}
+      >
+        <Grid item xs={1}>
+          {label}:
+        </Grid>
+        <Grid item xs={11}>
+          <Slider
+            value={val}
+            onChange={(e, v) => setVal(v)}
+            valueLabelDisplay="auto"
+            valueLabelFormat={v =>
+              `${truncateDecimalNumber(label === 'r' ? v : v / Math.PI, 4)}${
+                label === 'r' ? '' : 'π'
+              }`
```

Comment:
Possible to hoist this up into a function? hard for me to see what's going on. Not sure what r is.

```javascript
function formatter(label) 
  const isRadius  = label === 'r'; // useful variable name
  return (value) => {
    const val = truncateDecimalNumber(isRadius ? : v / Math.PI, 4);
    ... 
  }
}

//... 

valueLabelFormat={formatter(label)}
```